pkg/preparer/orchestrate_test.go

```diff
@@ -127,28 +169,32 @@ func TestPreparerFailsIfInstallFails(t *testing.T) {
 	}
 	newManifest := testManifest(t)
 
-	p := testPreparer(&FakeStore{})
+	p, hooks := testPreparer(&FakeStore{})
 	success := p.installAndLaunchPod(newManifest, testPod, logging.DefaultLogger)
 
 	Assert(t).IsFalse(success, "The deploy should have failed")
+	Assert(t).IsTrue(hooks.ranBeforeInstall, "should have ran before_install hooks")
 	Assert(t).IsTrue(testPod.installed, "Install should have been attempted")
 	Assert(t).IsFalse(testPod.launched, "Launch should not have happened")
+	Assert(t).IsFalse(hooks.ranAfterLaunch, "should not have run after_launch hooks")
 }
 
-func TestPreparerWillNotLaunchIfSHAIsTheSame(t *testing.T) {
+func TestPreparerWillNotInstallOrLaunchIfSHAIsTheSame(t *testing.T) {
 	testManifest := testManifest(t)
 	testPod := &TestPod{
 		currentManifest: testManifest,
 	}
 
-	p := testPreparer(&FakeStore{
+	p, hooks := testPreparer(&FakeStore{
 		currentManifest: testManifest,
 	})
 	success := p.installAndLaunchPod(testManifest, testPod, logging.DefaultLogger)
 
 	Assert(t).IsTrue(success, "Should have been a success to prevent retries")
+	Assert(t).IsFalse(hooks.ranBeforeInstall, "Should not have run hooks prior to install")
+	Assert(t).IsFalse(testPod.installed, "Should not have installed")
 	Assert(t).IsFalse(testPod.launched, "Should not have attempted to launch")
-	Assert(t).IsTrue(testPod.installed, "Should have installed")
```

Comment:
This is the only test that changed semantically - Installs should not happen if the SHA is different.